src/input/vita.c

```diff
@@ -442,23 +448,23 @@ void vitainput_config(CONFIGURATION config) {
   map.abs_rx = 2;
   map.abs_ry = 3;
 
-  map.btn_south = SCE_CTRL_CROSS - 1;
-  map.btn_east = SCE_CTRL_CIRCLE - 1;
-  map.btn_north = SCE_CTRL_TRIANGLE - 1;
-  map.btn_west = SCE_CTRL_SQUARE - 1;
-  map.btn_select = SCE_CTRL_SELECT - 1;
-  map.btn_start = SCE_CTRL_START - 1;
-  map.btn_thumbl = SCE_CTRL_LTRIGGER - 1;
-  map.btn_thumbr = SCE_CTRL_RTRIGGER - 1;
-  map.btn_dpad_up = SCE_CTRL_UP - 1;
-  map.btn_dpad_down = SCE_CTRL_DOWN - 1;
-  map.btn_dpad_left = SCE_CTRL_LEFT - 1;
-  map.btn_dpad_right = SCE_CTRL_RIGHT - 1;
-
-  map.btn_tl = TOUCHSEC_NORTHWEST;
-  map.btn_tr = TOUCHSEC_NORTHEAST;
-  map.btn_tl2 = TOUCHSEC_SOUTHWEST;
-  map.btn_tr2 = TOUCHSEC_SOUTHEAST;
+  map.btn_south       = SCE_CTRL_CROSS      | INPUT_TYPE_GAMEPAD;
+  map.btn_east        = SCE_CTRL_CIRCLE     | INPUT_TYPE_GAMEPAD;
+  map.btn_north       = SCE_CTRL_TRIANGLE   | INPUT_TYPE_GAMEPAD;
+  map.btn_west        = SCE_CTRL_SQUARE     | INPUT_TYPE_GAMEPAD;
+  map.btn_select      = SCE_CTRL_SELECT     | INPUT_TYPE_GAMEPAD;
+  map.btn_start       = SCE_CTRL_START      | INPUT_TYPE_GAMEPAD;
+  map.btn_thumbl      = SCE_CTRL_LTRIGGER   | INPUT_TYPE_GAMEPAD;
+  map.btn_thumbr      = SCE_CTRL_RTRIGGER   | INPUT_TYPE_GAMEPAD;
+  map.btn_dpad_up     = SCE_CTRL_UP         | INPUT_TYPE_GAMEPAD;
+  map.btn_dpad_down   = SCE_CTRL_DOWN       | INPUT_TYPE_GAMEPAD;
+  map.btn_dpad_left   = SCE_CTRL_LEFT       | INPUT_TYPE_GAMEPAD;
+  map.btn_dpad_right  = SCE_CTRL_RIGHT      | INPUT_TYPE_GAMEPAD;
+
+  map.btn_tl          = TOUCHSEC_NORTHWEST  | INPUT_TYPE_TOUCHSCREEN;
+  map.btn_tr          = TOUCHSEC_NORTHEAST  | INPUT_TYPE_TOUCHSCREEN;
```

Comment:
minus calculation looks little mess and in my test, raised conflict between touchscreen button.
so i append input type mark